### PR TITLE
038 - localStorage functions

### PR DESCRIPTION
**Pull request by MarekTalpas**
I made 2 functions - one for setting and another one for getting progress times of videos using local storage and also added appropriate tests.

### CODE REVIEW

File: src/api/local-storage.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+export const HISTORY_LIST: string = 'history';
+
+export function saveVideoProgress(id: number, progress: number) {
+  if (typeof (localStorage) !== 'undefined') {
+    const currentHistory: ?string = localStorage.getItem(HISTORY_LIST);
+    const videoProgressObject: VideoProgressType = { episodeId: id, progressTime: progress };
+    if (currentHistory) {
+      // $FlowFixMe
+      const indexOfVideo = currentHistory.findIndex(element => element.episodeId === id);
```
Comment (varholak-peter) [inline src/api/local-storage.js]: please provide a reason in the PR why `$FlowFixMe` is needed here

File: src/api/local-storage.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      if (indexOfVideo === -1) {
+        // $FlowFixMe
```
Comment (varholak-peter) [inline src/api/local-storage.js]: as above

File: src/api/local-storage.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        currentHistory.push(videoProgressObject);
+        localStorage.setItem(HISTORY_LIST, JSON.stringify(currentHistory));
+      } else {
+        // $FlowFixMe
+        currentHistory.splice(indexOfVideo, 1, videoProgressObject);
+        localStorage.setItem(HISTORY_LIST, JSON.stringify(currentHistory));
+      }
+    } else {
+      localStorage.setItem(HISTORY_LIST, JSON.stringify(videoProgressObject));
+    }
```
Comment (varholak-peter) [inline src/api/local-storage.js]: There is a lot of duplicity with setItem here, I would prefer to have only one setItem at the end of function with a mutable object here.

File: src/api/local-storage.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    if (currentHistory) {
+      // $FlowFixMe
+      const indexOfVideo = currentHistory.findIndex(element => element.episodeId === id);
+      if (indexOfVideo === -1) {
+        // $FlowFixMe
+        currentHistory.push(videoProgressObject);
+        localStorage.setItem(HISTORY_LIST, JSON.stringify(currentHistory));
+      } else {
+        // $FlowFixMe
+        currentHistory.splice(indexOfVideo, 1, videoProgressObject);
```
Comment (varholak-peter) [inline src/api/local-storage.js]: This is unnecessary complex.

File: src/api/local-storage.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    } else {
+      localStorage.setItem(HISTORY_LIST, JSON.stringify(videoProgressObject));
+    }
+  }
+}
+
+export function getProgressTimeById(id: number) {
+  const data: ?string = localStorage.getItem(HISTORY_LIST);
+  if (data) {
+    // $FlowFixMe
```
Comment (varholak-peter) [inline src/api/local-storage.js]: as above

File: src/api/local-storage.js (hunk 1 above)
Comment (MarekTalpas) [inline src/api/local-storage.js]: flow automatically convert outcome of findIndex function to object, but we need indexOfVideo variable to be a number, so it causes undesirable interference.

**Pull request merged**